db/dcrpg/queries.go

```diff
@@ -2983,76 +2983,123 @@ func retrieveTicketsPriceByHeight(ctx context.Context, db *sql.DB, interval int6
 func retrieveCoinSupply(ctx context.Context, db *sql.DB, timeArr []dbtypes.TimeDef,
 	sumArr, estimateArr []float64, params *chaincfg.Params) ([]dbtypes.TimeDef,
 	[]float64, []float64, error) {
-	var since, initialTime time.Time
-	var sum, estimate, preminedCoins float64
-
-	var reward = dcrutil.Amount(params.BaseSubsidy).ToCoin()
-	var targerBlockTime = float64(params.TargetTimePerBlock)
-	var inflationRate = float64(params.MulSubsidy) / float64(params.DivSubsidy)
-	var subsidyReduction = float64(params.SubsidyReductionInterval)
-
-	if c := len(timeArr); c > 0 && len(sumArr) == c && len(estimateArr) == c {
+	var sum, estimate float64
+	var since, premineTime time.Time
+
+	premineBlockInd := 1
+	reward := dcrutil.Amount(params.BaseSubsidy).ToCoin()
+	targetBlockTime := params.TargetTimePerBlock.Minutes()
+	inflationRate := float64(params.MulSubsidy) / float64(params.DivSubsidy)
+	subsidyReduction := float64(params.SubsidyReductionInterval)
+
+	// https://github.com/decred/dcrd/blob/5076a00512a521cea3c51b443b50970804dbe712/blockchain/subsidy_test.go#L52-L54
+	maxCoinSupply := 20999999.99800912
```

Comment:
Only true for mainnet.  But a quick call to `txhelpers.UltimateSubsidy(params)` will give you the number too (in atoms).